extensions/wikia/FacebookClient/SpecialFacebookConnectController.class.php

```diff
@@ -1,280 +0,0 @@
-			return true;
-		}
-
-		// Setup the session as is done when a request first starts
-		if ( !$wg->SessionStarted ) {
-			wfSetupSession();
-		}
-		$user->setCookies();
-
-		// Store the user in the global user object
-		$wg->User = $user;
-
-		$this->forward( __CLASS__, 'successfulConnect' );
-
-		\FacebookClientHelper::track( 'facebook-link-existing' );
-	}
-
-	/**
-	 * Connect the already logged in Wikia user to a Facebook account.  By the time they get here
-	 * they should already have logged into Facebook and have a Facebook user ID.
-	 *
-	 * @throws FacebookMapModelInvalidDataException
-	 */
-	public function connectCurrentUser() {
-		$wg = $this->wg;
-
-		$fb = FacebookClient::getInstance();
-		$fbUserId = $fb->getUserId();
-
-		// The user must be logged into Facebook.
-		if ( !$fbUserId ) {
-			$wg->Out->showErrorPage( 'fbconnect-error', 'fbconnect-errortext' );
-			$this->skipRendering();
-			return true;
-		}
-
-		// Create user mapping
-		$status = $this->fbClientFactory->connectToFacebook( $wg->User->getId(), $fbUserId );
-		if ( ! $status->isGood() ) {
-			$this->showErrorPage( $status );
-			$this->skipRendering();
-			return true;
-		}
-
-		$this->forward( __CLASS__, 'successfulConnect' );
-
-		\FacebookClientHelper::track( 'facebook-link-existing' );
-	}
-
-	/**
-	 * This code is run when the user first reaches the Special:FacebookConnect page.  It displays
-	 * a login form.  Successful login will connect that user to their Facebook account.
-	 *
-	 * @throws MWException
-	 * @throws ReadOnlyError
-	 */
-	public function connectUser() {
-		$wg = $this->wg;
-
-		$title = wfMessage('fbconnect-connect-existing')->plain();
-		$this->getContext()->getOutput()->setPageTitle( $title );
-
-		$titleObj = SpecialPage::getTitleFor( 'FacebookConnect' );
-		if ( wfReadOnly() ) {
-			$wg->Out->readOnlyPage();
-			return false;
-		} elseif ( $wg->User->isBlockedFromCreateAccount() ) {
-			// Add code to handle blocking here.
-
-			// This was the old call and despite the comment, the method makes heavy use of $this
-			//LoginForm::userBlockedMessage($wg->User->getBlock()); //this is not an explicitly static method but doesn't use $this and can be called like static (fixes RT#75589)
-			return false;
-		} elseif ( count( $permErrors = $titleObj->getUserPermissionsErrors( 'createaccount', $wg->User, true ) ) > 0 ) {
-			$wg->Out->showPermissionsErrorPage( $permErrors, 'createaccount' );
-			return false;
-		}
-
-		// Grab the UserName from the cookie if it exists
-		$userNameCookie = $wg->CookiePrefix . 'UserName';
-		$userName = isset( $_COOKIE[$userNameCookie] ) ? trim( $_COOKIE[$userNameCookie] ) : '';
-
-		$errorMessage = $wg->Request->getVal( 'errorMessage' );
-
-		if ( $errorMessage ) {
-			$this->errorMessage = wfMessage( $errorMessage )->plain();
-		}
-		$this->formAction =  SpecialPage::getSafeTitleFor( 'FacebookConnect' )->getLocalURL();
-		$this->legendText = wfMessage( 'fbconnect-connect-login-legend' )->text();
```

Comment:
And `fbconnect-connect-username-label` ... and `fbconnect-connect-password-label`